clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -475,11 +500,20 @@ public Node leastLoadedNode(long now) {
     }
 
     public static AbstractResponse parseResponse(ByteBuffer responseBuffer, RequestHeader requestHeader) {
+        return parseResponseMaybeUpdateThrottleTimeMetrics(responseBuffer, requestHeader, null);
+    }
+
+    private static AbstractResponse parseResponseMaybeUpdateThrottleTimeMetrics(ByteBuffer responseBuffer, RequestHeader requestHeader, Sensor throttleTimeSensor) {
         ResponseHeader responseHeader = ResponseHeader.parse(responseBuffer);
         // Always expect the response version id to be the same as the request version id
         ApiKeys apiKey = ApiKeys.forId(requestHeader.apiKey());
         Struct responseBody = apiKey.parseResponse(requestHeader.apiVersion(), responseBuffer);
         correlate(requestHeader, responseHeader);
+        if (throttleTimeSensor != null) {
+            String throttleTimeFieldName = "throttle_time_ms";
```

Comment:
@ijuma Can you take a look at the latest commits before I merge? I also had to add checkstyle exempt rules to build after rebasing. The build failure looks unrelated. Thank you...